### PR TITLE
[#691] Support tag_ids filter queries

### PR DESCRIPTION
**Pull request by lucapette**

### CODE REVIEW

File: backend/api/communication/src/main/java/co/airy/core/api/communication/lucene/DocumentMapper.java [hunk 1]
```diff
@@ -30,7 +33,13 @@ public Document fromConversationIndex(ConversationIndex conversation) {
         document.add(new StoredField("unread_message_count", conversation.getUnreadMessageCount()));
 
         for (Map.Entry<String, String> entry : conversation.getMetadata().entrySet()) {
-            document.add(new TextField("metadata." + entry.getKey(), entry.getValue(), Field.Store.YES));
+            if (!entry.getKey().equals(MetadataKeys.TAGS)) {
+                document.add(new TextField("metadata." + entry.getKey(), entry.getValue(), Field.Store.YES));
+            }
+        }
```
Comment (lucapette) [inline backend/api/communication/src/main/java/co/airy/core/api/communication/lucene/DocumentMapper.java:39]: I did this for completeness but to be honest I would remove the indexing all together for the metadata and rework this part when we know what kind of queries we'd need to support

Comment (chrismatix) [reply backend/api/communication/src/main/java/co/airy/core/api/communication/lucene/DocumentMapper.java:39]: Agreed

Comment (paulodiniz) [reply backend/api/communication/src/main/java/co/airy/core/api/communication/lucene/DocumentMapper.java:39]: 👍

File: backend/api/communication/src/main/java/co/airy/core/api/communication/dto/Conversation.java [hunk 2]
```diff
@@ -55,6 +58,15 @@ private String prettifySource(String source) {
         return capitalize(source);
     }
 
+    @JsonIgnore
+    public List<String> getTagIds() {
```
Comment (chrismatix) [inline backend/api/communication/src/main/java/co/airy/core/api/communication/dto/Conversation.java:62]: Good refactor 👍

**Review by chrismatix — APPROVED** [review]

**Review by chrismatix — APPROVED** [review]

**Pull request merged**